_posts/2021-08-16-Day-6-Post.md

```diff
@@ -0,0 +1,59 @@
+---
+layout: post
+title: "Doing more? Say Less. we have C++"
+author: Kevin Fong
+---
+## C++ as a Programming Language
+
+C++ is a general purpose programming language created as an extension of the C programming language. It was designed with an orientation toward system programming and embedded, resource-contained software and large systems, with performance, efficiency, and flexibility of us as its design highlights.
+
+C++ is an object oriented, staticly typed, compiled language. As a static typed language, all objects have their data type declared upon definition. As a compiled language, the source code is translated using compilers to produce machine code. Once compiled, the machine code is often undecipherable to humans. Because the compilation is performed in one step, compilation errors will prevent the code from compiling, and the code must debugged before the code is run and executed.
+
+## Comparing Python and C++"
+
+Python by comparison is a high-level, mutli-paradigm, dynamically typed, garbage-collected, interpreted language. As an interpreted language, Python code is not compiled into machine instructions and is ready to be read and interpreted as is. Because it is dynamically written, data types are not declared, making Python more flexible and easy to use. However C++ is able to perform more effciently because all data types are declared at the point of compilation, so the machine does not have to consume any extra time to interpret the code to determine what data types are being used.
```

Comment:
```suggestion
Python by comparison is a high-level, mutli-paradigm, dynamically typed, garbage-collected, interpreted language. As an interpreted language, Python code is not compiled into machine instructions and is ready to be read and interpreted as is. Because it is dynamically written, data types are not declared, making Python more flexible and easy to use. However C++ is able to perform more efficiently because all data types are declared at the point of compilation, so the machine does not have to consume any extra time to interpret the code to determine what data types are being used.
```